### PR TITLE
Make footer sticky by adding flexbox container around nav, body, and …

### PR DESCRIPTION
**Pull request by appleJax**
# What's in this PR?

Adds a .flexbox-container div around the nav, body, and footer to make footer stick to bottom of screen.

Also adds a .flexbox-spacer div beween the body and footer to make the body appear as normal (without the spacer, the body is centered between the nav and footer, leaving more space between the nav and body)

Styles for .flexbox-container:
- flex-direction: column
- justify-content: space-between
- min-height: 100vh

Styles for .flexbox-spacer:
- flex-grow: 2

Before:

screen shot 2016-09-18 at 6 45 17 pm

After:

screen shot 2016-09-18 at 7 05 16 pm

Alternate option without spacer (NOT PART OF THIS PR, SHOWN ONLY FOR COMPARISON):

screen shot 2016-09-18 at 6 43 02 pm

### CODE REVIEW

File: app/styles/main.scss [hunk 1]
```diff
@@ -23,6 +23,17 @@ html, body {
...
 
+.flexbox-container {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  min-height: 100vh;
+}
+
+.flexbox-spacer {
+  flex-grow: 2;
```
Comment (JoshSmith) [inline app/styles/main.scss]: What does `flex-grow: 2` mean?

Comment (appleJax) [inline app/styles/main.scss]: Flex grow is a ratio that controls how much space an element takes up on the page. `flex-grow: 2` means that div can expand to take up twice as much space as the other elements on the page.

Comment (JoshSmith) [inline app/styles/main.scss]: Why is that needed in this case? I mean why, specifically, twice as much?

Comment (JoshSmith) [inline app/styles/main.scss]: Will `flex: auto;` not do the same? Serious flexbox ignorance here.

Comment (appleJax) [inline app/styles/main.scss]: Since the `min-height` of the `.flexbox-container` is 100% of the screen height, the div takes up no space unless the content doesn't fill the screen. If the content does not fill the screen, the `.flexbox-spacer` div will expand, pushing the footer to the bottom of the page, and the body towards the nav.

Comment (appleJax) [inline app/styles/main.scss]: The more I compare the different versions, the more I like the alternate version though, the one where the body is centered between the nav and footer.

**Comment by appleJax** [comment]
The alternate version looked weird at first with so much space between the nav and body, but I think the user is going to be focused on the content of the body (in this case, the sign-up form), so it actually makes sense to have it sit in the center of the screen.

File: app/styles/main.scss (hunk 1 above)
Comment (appleJax) [inline app/styles/main.scss]: `flex` is a shorthand for `flex-grow, flex-shrink, flex-basis`, so I don't think `flex: auto` is valid syntax

Comment (appleJax) [inline app/styles/main.scss]: When I was testing `flex-grow`, any value greater than 1 resulted in the same output on screen, so I stuck with '2' arbitrarily because it was the nearest whole number greater than 1

Comment (appleJax) [inline app/styles/main.scss]: After getting used to the alternate version, though, I honestly think we should get rid of the spacer.

Comment (JoshSmith) [inline app/styles/main.scss]: @appleJax https://css-tricks.com/almanac/properties/f/flex/#article-header-id-3

Comment (JoshSmith) [inline app/styles/main.scss]: I disagree on the alternate version. On other pages it will look particularly odd. This is a kind of edge case.

Comment (appleJax) [inline app/styles/main.scss]: Oh, good tip on `flex: auto`. Let me try it real quick...

Comment (appleJax) [inline app/styles/main.scss]: So yeah, looks like `flex: auto` has the same result as `flex-grow: 2` in this case

Comment (JoshSmith) [inline app/styles/main.scss]: Yeah, I would change to that. It looks clearer to me and does what it says: fills the space completely.

Comment (appleJax) [inline app/styles/main.scss]: Agreed, I didn't know about that trick or I would have used it.

Comment (appleJax) [inline app/styles/main.scss]: I still think you need to give the centered layout more time though. I totally understand how weird it looks at first glance, and it must be way weirder for you since you've seen it the other way for much longer than I have. I really think the body is king though, and the user knows where to find the nav if they need it.

Comment (JoshSmith) [inline app/styles/main.scss]: Awesome! We both learned something.

Comment (JoshSmith) [inline app/styles/main.scss]: I'm mostly referring to other pages (and designs I have in my head right now, which will be made more difficult by such a global style).

Comment (appleJax) [inline app/styles/main.scss]: Can you elaborate a little? What kind of designs are you thinking about? And which specific pages?

File: app/styles/main.scss [hunk 2]
```diff
@@ -23,6 +23,17 @@ html, body {
   height: 100%;
 }
 
+.flexbox-container {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  min-height: 100vh;
```
Comment (JoshSmith) [inline app/styles/main.scss]: What does `100vh` mean?

Comment (appleJax) [inline app/styles/main.scss]: 100% of view height. I should look that up actually, lol, not sure how it's different from 100%.

Comment (appleJax) [inline app/styles/main.scss]: I think vh is a percent relative to the screen, and % is based on parent container. So maybe %   is more clear in this case?

Comment (JoshSmith) [inline app/styles/main.scss]: I don't know enough to say for sure.

**Comment by appleJax** [comment]
Added `padding-top: 10vh` to signup-form.scss and `padding-top: 15vh` to login-form.scss to give some breathing room to the forms on those pages. Tried percentages which looked fine full-screen, but the padding did not shrink with a limited view height.

screen shot 2016-09-20 at 9 29 38 am

screen shot 2016-09-20 at 9 29 55 am

File: app/styles/components/login-form.scss [hunk 3]
```diff
@@ -1,4 +1,5 @@
 .login-form {
   @include span-columns(5);
   @include shift(3.5);
+  padding-top: 15vh;
```
Comment (JoshSmith) [inline app/styles/components/login-form.scss]: Would prefer `10vh` here to make it consistent.

Comment (appleJax) [inline app/styles/components/login-form.scss]: Ok
